language-server/test/dotty/tools/languageserver/util/embedded/CodeMarker.scala

```diff
@@ -0,0 +1,44 @@
+package dotty.tools.languageserver.util.embedded
+
+import dotty.tools.languageserver.util.server.TestFile
+import dotty.tools.languageserver.util.{CodeRange, PositionContext}
+
+import org.eclipse.lsp4j._
+
+import PositionContext._
+
+/** Used to mark positions in the code */
+class CodeMarker(val name: String) extends Embedded {
+
+  def to(other: CodeMarker): CodeRange = CodeRange(this, other)
+
+  def file: PosCtx[TestFile] = implicitly[PositionContext].positionOf(this)._1
```

Comment:
You could add `implicit def posCtx(implicit ctx: PositionContext): PositionContext = ctx` in the class to be able to do `posCtx.positionOf ...`